src/controllers/rankings.js

```diff
@@ -0,0 +1,66 @@
+const { graphql } = require('graphql');
+const { schema } = require('./graphql')
+
+const query = `
+query UserRankings($user_id: Int) {
+  userRankings(prev_ranking_idIsNull: true, user_id: $user_id) {
+    ...RankingFields
+    ...RankingsRecursive
+  }
+}
+
+fragment RankingsRecursive on UserRankings {
+  nextRanking {
+    ...RankingFields
+    nextRanking {
+      ...RankingFields
+      nextRanking {
+        ...RankingFields
+        nextRanking {
+          ...RankingFields
+          nextRanking {
+            ...RankingFields
+            nextRanking {
+              ...RankingFields
+              nextRanking {
+                ...RankingFields
+                nextRanking {
+                  ...RankingFields
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+fragment RankingFields on UserRankings {
+  ranking_id: id
+  item {
+    id
+    item_name
+    item_details {
+      description
+    }
+  }
+}
+`
+
+async function getUserRankings(ctx, next) {
+  const { data } = await graphql(schema, query, null, null, {user_id: ctx.state.user.id})
+  const flattenedUserRankings = flattenRankings(data.userRankings[0], [])
+  ctx.body = flattenedUserRankings
+}
+
+function flattenRankings(currentItem, rankings) {
+  if (!currentItem.nextRanking) {
+    return [currentItem.item, ...rankings]
+  } else {
+    rankings = [currentItem.item, ...rankings]
```

Comment:
It might be worth `omit`ing `nextRanking` from `currentItem.item`, since that information is being transformed into array order.

Reply:
I thought you would be impressed with my use of recursion here

Reply:
![](https://media.giphy.com/media/OK27wINdQS5YQ/giphy.gif)

Reply:
Tail recursive, even

Reply:
Actually, I'll just use this function if we add that `omit`...I've got a totally-not-nearly-as-good-but-functionally-identical version of this in my branch.

Reply:
Also we need to somehow embed this gif as a comment in the code, so if you could get around to that that'd be great.

Reply:
I'm a little confused about the `omit`. `currentItem` has two keys `item` and `nextRanking`. I'm only spreading `currentItem.item` so `nextRanking` is already being omitted.

Reply:
Oh nice. I got confused with my having to do it for the objection version.